lib/v2/finology/utils.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable linebreak-style */
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+const getItems = async (ctx, url, extra) => {
+    const mainUrl = 'https://insider.finology.in';
+    const { data: response } = await got(url);
+    const $ = cheerio.load(response);
+    const listItems = $(extra.selector)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+            const title = item.find('p.text-m-height').text();
+            const link = item.find('a').attr('href');
+            const pubDate = extra.date ? parseDate(item.find('div.text-light p').first().text(), 'DD-MMM-YYYY') : '';
+            const itunes_item_image = item.find('img').attr('src');
+            const category = item.find('p.pt025').text();
+            return {
+                title,
+                link: `${mainUrl}${link}`,
+                pubDate,
+                itunes_item_image,
+                category,
+            };
+        });
+
+    const items = await Promise.all(
+        listItems.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: response } = await Promise.race([
+                    got(item.link),
+                    new Promise((resolve) => {
+                        setTimeout(
+                            () => resolve({ data: `<div id="main-wrapper"><div id="insiderhead"><h1>Website did not load within Timeout Limits. <a href="${item.link}">Check with Website if the page is slow</a></h1></div></div>` }),
+                            30000
+                        );
+                    }),
```

Comment:
RSSHub has a configurable timeout of `REQUEST_TIMEOUT` and retry of`REQUEST_RETRY`. Requests may end before or after this fixed timeout.
Besides that, the cache will store a polluted value instead of not storing.

Reply:
The problem i m facing here is :
One of the articles loads very slowly(idk why), hence RSSHub times out and makes the whole Feed time out, that's why i used this. Please guide me if there is a better way

Reply:
You can use [Promise.allSettled()](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) with `filter()` and `map()`.